app/src/main/java/app/revanced/manager/ui/viewmodel/AppSelectorViewModel.kt

```diff
@@ -17,15 +18,19 @@ import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
 import java.io.File
 import java.nio.file.Files
 
 class AppSelectorViewModel(
     private val app: Application,
     private val pm: PM,
     private val patchBundleRepository: PatchBundleRepository
-) : ViewModel() {
-    private val inputFile = File(app.cacheDir, "input.apk").also {
+) : ViewModel(), KoinComponent {
+    private val fs: Filesystem by inject()
+
+    private val inputFile = File(fs.tempDir, "input.apk").also {
```

Comment:
tempDir is only valid for the duration of the app process, but this needs to live longer than that. Use getDir instead.